auslib/test/blobs/test_systemaddons.py

```diff
@@ -222,3 +222,26 @@ def testDoesNotContainForbiddenDomain(self):
 """)
         self.assertFalse(blob.containsForbiddenDomain('gg',
                                                       self.whitelistedDomains))
+
+    def testIncorrectAddonLayout(self):
+        blob = SystemAddonsBlob()
+        blob.loadJSON("""
+    {
+    "name": "fake",
+    "schema_version": 1000,
+    "hashFunction": "SHA512",
+    "addons": {
+        "c": {
+            "version": "1",
+            "platforms": {
+                "p": {
+                    "filesize": 2,
+                    "hashValue": "3",
+                    "fileUrl": "http://a.com/blah"
+                }
+            }
+        }
+    }
+}
+""")
+        blob.validate('gg', self.whitelistedDomains)
```

Comment:
@tyagi-iiitv  . you're validaiting an invalid blob. So when you're invoking blob.validate method it is throwing BlobValidationError as expected. Simply place that part in a try catch and assert that the caught exception and message are as expected.